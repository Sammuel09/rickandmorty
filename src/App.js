import { useState, useEffect } from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import Header from "./components/Header";
import Footer from "./components/Footer";
import Error from "./components/Error";
import SingleCharacter from "./views/SingleCharacter";
import { request } from "./utils/apiRequest";
import Home from "./views/Home";

function App() {
  const [characters, setCharacter] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    const getCharacters = async () => {
      setHasError(false);
      setIsLoading(true);
      try {
        const charactersFromApi = await request("character");
        setCharacter(charactersFromApi.results);
      } catch (error) {
        setHasError(true);
      } finally {
        setIsLoading(false);
      }
    };
    getCharacters();
  }, []);

  return (
    <Router>
      <div className="container">
        <Header />
        {hasError && <Error />}
        <Route
          path="/"
          exact
          render={() => <Home isLoading={isLoading} characters={characters} />}
        />
        <Route
          exact
          path="/character/:characterId"
          component={SingleCharacter}
        />
        <Footer />
      </div>
    </Router>
  );
}

export default App;
