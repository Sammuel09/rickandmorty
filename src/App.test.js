import { render } from "@testing-library/react";
import App from "./App";

test("renders the App component", () => {
  render(<App />);
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
