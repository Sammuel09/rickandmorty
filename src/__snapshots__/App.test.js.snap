// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="header solid"
    >
      <img
        alt="Rick and Morty Logo"
        src="rickandmorty.svg"
      />
    </div>
    <div
      class="body-container"
    >
      <div
        class="spinner-container"
      >
        <div
          class="lds-ring"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      Built with ‚ù§Ô∏è and üí°
    </div>
  </div>
</DocumentFragment>
`;
